src/Action.js

```diff
@@ -0,0 +1,28 @@
+export default class Action {
+	constructor (promise) {
+		this.promise = promise
+	}
+
+	// default onFulfilled action
+	/* istanbul ignore next */
+	fulfilled (p) {
+		this.promise._become(p)
+	}
+
+	// default onRejected action
+	rejected (p) {
+		this.promise._become(p)
+		return false
+	}
+
+	tryCall (f, x) {
```

Comment:
Ah, ok, thanks for explaining your thinking.  It's not really clear to me what is the right thing to do with an exception from this.handle.  I'd be ok with letting the exception manifest as a rejection, but I find myself really liking that at least there's a distinction being made now.

The code makes me cringe a little, but I can't come up with anything better!

So :+1: 😄